src/linpde_gp/linfunctls/_flattened.py

```diff
@@ -0,0 +1,32 @@
+import functools
+
+import numpy as np
+import probnum as pn
+
+from ._linfunctl import LinearFunctional
+
+class FlattenedLinearFunctional(LinearFunctional):
```

Comment:
I still find `FlattenedLinearFunctional` slightly more intuitive because even with a different order, the goal is still to turn objects with multiple dimensions into flat objects with a single dimension. But I'm also fine with `ReshapedLinearFunctional` :)